tinygrad/runtime/ops_clang.py

```diff
@@ -1,21 +1,46 @@
 from typing import Optional, List
-import ctypes, subprocess, pathlib, tempfile
+import platform, struct, ctypes, subprocess, pathlib, tempfile, tinygrad.runtime.autogen.libc as libc
+from mmap import PROT_READ, PROT_WRITE, PROT_EXEC, MAP_ANON, MAP_PRIVATE
+from tinygrad.runtime.support.elf import elf_loader
 from tinygrad.device import Compiled, Compiler, MallocAllocator
-from tinygrad.helpers import cpu_time_execution, DEBUG, cpu_objdump
+from tinygrad.helpers import cpu_time_execution, DEBUG, OSX, getenv, cpu_objdump
 from tinygrad.renderer.cstyle import ClangRenderer
 
+MAP_JIT, global_handle = 0x0800 if OSX else 0, ctypes.CDLL(None)
```

Comment:
https://man7.org/linux/man-pages/man3/dlopen.3.html

> If filename is NULL, then the returned handle is for the main
> program.  When given to [dlsym(3)](https://man7.org/linux/man-pages/man3/dlsym.3.html), this handle causes > a search for
> a symbol in the main program, followed by all shared objects
> loaded at program startup, and then all shared objects loaded by
> dlopen() with the flag RTLD_GLOBAL.

Reply:
I thought about using autogen_stubs on macos in separate runner to generate bindings for mac-only libc functions but macos headers use a lot of syntax and features that aren't supported by clang2py